source/content/guides/drupal-9-migration/06-build-tools-to-d9-build-tools.md

```diff
@@ -0,0 +1,276 @@
+---
+title: Build Tools to D9 with Build Tools
+subtitle: subtitle
+description: Description
+categories: [develop]
+cms: drupal-9
+tags: [code, launch, migrate, site, updates]
+contributors: [stovak]
+reviewed: "2021-05-15"
+layout: guide
+showtoc: true
+permalink: docs/guides/drupal-9-migration/build-tools-to-d9-build-tools
+anchorid: drupal-9-migration/build-tools-to-d9-build-tools
+editpath: drupal-9-migration/06-build-tools-to-d9-build-tools.md
+---
+
+In this doc, you'll create a new Drupal 9 site and migrate the code from an existing Drupal 8 site to it.
+
+## Will This Guide Work for Your Site?
+
+<Partial file="drupal-9/upgrade-site-requirements.md" />
+
+## Prepare the Local Environment
+
+<Partial file="drupal-9/prepare-local-environment.md" />
+
+This doc uses the following aliases:
+
+- **Alias:** alias
+
+## Build Tools Workflow
+
+This tutorial is for you if you meet the following criterion:
+
+1. You have an external repository outside of pantheon that manages your site's code.
+
+1. You use a build service like Github Actions, Circle CI or Travis to "build" the site.
+
+1. Those build artifacts are pushed to your pantheon repo.
+
+1. Your site has a blue banner and is database-ready for the upgrade to d9
+
+## Tools you will need
+
+Terminus, Bash JSON Query, rsync
+
+```bash{promptUser: user}
+brew install terminus jq rsync
+```
+
+## Wagons HO!
+
+- Clone your GitHub source repo and `cd` into the directory, then:
+
+```bash{promptUser: user}
+git checkout -b d9-upgrade-2021
+```
+
+- Let's export the configs from production to make sure our local version has the latest:
+
+```bash{promptUser: user}
+# Export the latest from the live environment
+# to sites/default/files/config
+terminus drush gk-8.live -- config:export \
+   --destination sites/default/files/config
+
+# Use terminus to get the (read-only) SFTP command
+# specific to the live environment.
+SFTP_COMMAND=$(terminus connection:info gk-8.live \
+   --format=json | jq -r ".sftp_command")
+
+# For rysnc, all we really need is the crazy-long
+# user and host name. Strip out the rest.
+# (Make sure the empty space is there at the end
+# before the bracket!)
+RSYNC_HOST=$(terminus connection:info {Site Name}.live --field=sftp_host)
+
+# Use that hostname to do an rsync from that recent
+# config:export
+rsync -rvlz --copy-unsafe-links --size-only --checksum \
+   --ipv4 --progress -e 'ssh -p 2222' \
+   "${RSYNC_HOST}:files/config" .
+```
+
+   If you do a `git status` it should show changed files in the `config` directory if there are any changed configs in production.
+
+### Let's start ugradin' stuff!
+
+```bash{promptUser: user}
+   composer require \
+     drupal/upgrade_status:^3 \
+     drupal/devel:^4 \
+     drush/drush:^10 \
+     -W --no-update
+   composer require \
+     phpunit/phpunit:"^8 | ^9 | ^10" \
+     phpstan/phpstan \
+     --dev -W --no-update
+```
+
```

Comment:
- Enable deep patching in cweagans/composer-patches:
```
composer config extra.enable-patching true
```
The project `pantheon-systems/drupal-integrations` now includes a patch that backports a bugfix from Drupal 9 to Drupal 8. This patch allows Drupal to display the correct version of your MariaDB server. If this patch is not installed, then your database version will always be reported as "Mysql 5.5.30". The Composer plugin `cweagans/composer-patches` will only install patches from dependencies if the `enable-patching` property is set to `true` in your composer.json file.